tool/tsh/common/db.go

```diff
@@ -1312,8 +1312,8 @@ func isDatabaseNameRequired(protocol string) bool {
 		return true
 	}
 	switch protocol {
-	case defaults.ProtocolOracle:
-		// Always require database name for the Oracle protocol.
+	// Always require database name for these protocols.
+	case defaults.ProtocolOracle, defaults.ProtocolSQLServer:
```

Comment:
@gabrielcorado @Tener do you recall if the CLI requires a database name for SQL server?  I put SQL server here to be backward compatible but google tells me sqlserver is fine without the database name.

Reply:
I vaguely recall you can just connect to the "server" without specifying the database name, so Google is likely correct here. We should probably test how it works 

Technically there is database-less mode for Oracle too, but it is a more restricted case and I don't think we need to support it; we may not be able to in fact.

Reply:
didn't get a chance to test this. the current code tries to be backwards compatible. lets revisit this during next release testing and make any changes necessary